### PR TITLE
Document the eth_accounts permission for snaps

### LINKED ISSUE #888: Document the eth_accounts permission for snaps
Document the first dynamic permission for snaps: `eth_accounts`

### PR DESCRIPTION
**Pull request by ziad-saab**
Fixes #888

### CODE REVIEW

File: snaps/reference/permissions.md [hunk 1]
```diff
@@ -179,3 +179,30 @@ Specify this permission in the manifest file as follows:
...
 },
 ```
+
+## Dynamic permissions
+
+Dynamic permissions are not requested in the manifest file. Instead, they are acquired during the lifecycle of the snap.
+
+## eth_accounts
+
+This dynamic permission is acquired when a snap calls the `eth_requestAccounts` method of the Ethereum provider. Calling `eth_requestAccounts` requires the [`ethereum-provider` endowment](#endowmentethereum-provider). The presence of the permission can be checked by calling [`wallet_getPermissions`](/wallet/reference/rpc-api/#wallet_getpermissions). If the permission is present, the return value of `wallet_getPermissions` will contain a permission with a `parentCapability` of `eth_accounts`. It will come with a caveat of `restrictReturnedAccounts`, an array of all the accounts that the user allowed for this snap:
```
Comment (FrederikBolding) [inline snaps/reference/permissions.md:189]: Worth mentioning that you should be able to request `eth_accounts` and/or `eth_requestAccounts`, which will also return the addresses without the extra permission caveats information?

File: snaps/reference/permissions.md [hunk 2]
```diff
@@ -179,3 +179,30 @@ Specify this permission in the manifest file as follows:
...
+      "value": [
+        "0xc403b37bf1e700cb214ea1be9de066824b420de6" // example connected account #1
+      ]
+    }
+  ],
+  "date": 1692616452846
+}
+```
+
+This permission can be revoked by the user by going to the "Connected sites" option in the MetaMask UI, and removing the connection from an account to the snap.
```
Comment (FrederikBolding) [inline snaps/reference/permissions.md:208]: This needs to be revoked in the specific snap's settings under snap permissions.

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: snaps/reference/permissions.md (hunk 1 above)
Comment (ziad-saab) [reply snaps/reference/permissions.md:189]: @FrederikBolding do you mean using `wallet_requestPermissions`? I didn't cover that in the PR, can't find any info on how to use it. My understanding is that calling `eth_accounts` won't popup the MetaMask window, so it won't generate any new permission. Can you elaborate on this please?

Comment (FrederikBolding) [reply snaps/reference/permissions.md:189]: Sorry, I mean after calling `eth_requestAccounts` to get the `eth_accounts` permission, you can simply call it again to get the already granted accounts. You don't need to use `wallet_getPermissions`. IIRC requesting `eth_accounts` should do the same. But fine by me to just reference `eth_requestAccounts`, it feels like a better pattern than `wallet_getPermissions`

File: snaps/reference/permissions.md [hunk 3]
```diff
@@ -179,3 +179,30 @@ Specify this permission in the manifest file as follows:
   "endowment:webassembly": {}
 },
 ```
+
+## Dynamic permissions
+
+Dynamic permissions are not requested in the manifest file. Instead, they are acquired during the lifecycle of the snap.
+
+## eth_accounts
```
Comment (Montoya) [inline snaps/reference/permissions.md:187]: This should be 3 `#` not 2. So it indents correctly.

File: snaps/reference/permissions.md (hunk 1 above)
Comment (Montoya) [reply snaps/reference/permissions.md:189]: This is a technical reference that documents the permissions. A separate page explaining how to use `eth_requestAccounts` would provide a best practice.

**Review by Montoya — CHANGES_REQUESTED** [review]
See requested change on line 187

**Review by hmalik88 — APPROVED** [review]

**Review by Montoya — APPROVED** [review]

**Review by alexandratran — APPROVED** [review]
Edited and made minor additions for clarity. LGTM.

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**